docs/contracts/constant-state-variables.rst

```diff
@@ -76,7 +76,8 @@ Immutable
 Variables declared as ``immutable`` are a bit less restricted than those
 declared as ``constant``: Immutable variables can be assigned an arbitrary
 value in the constructor of the contract or at the point of their declaration.
-They cannot be read during construction time and can only be assigned once.
+They can be assigned only once and can, from that point on, be read even during
+construction time.
```

Comment:
I'll add some more text here concerning what we define as initialization order WRT inline initialization and constructor.